cdap-proto/src/main/java/co/cask/cdap/proto/BusinessMetadataRecord.java

```diff
@@ -0,0 +1,97 @@
+/*
+ * Copyright 2015 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package co.cask.cdap.proto;
+
+import java.util.Objects;
+import javax.annotation.Nullable;
+
+/**
+ * Represent Busineess Metadata for CDAP.
+ */
+public class BusinessMetadataRecord {
+  private final String targetType;
+  private final Id.NamespacedId  targetId;
```

Comment:
The type of `targetId` is supposed to be the parent class for all target types, right? In that case let's just use `Id`, rather than `NamespacedId`. Using `NamespacedId` here is a bit misleading while reading the code. What do you say?